packages/devel/libtool/package.mk

```diff
@@ -34,3 +34,8 @@ PKG_AUTORECONF="no"
 
 PKG_CONFIGURE_OPTS_HOST="--enable-static --disable-shared"
 
+pre_build_host() {
+  cd $PKG_BUILD
+    autoreconf --force --verbose
```

Comment:
Why not set `PKG_AUTORECONF="yes"` instead? Although the real question is why this now needs autoreconf when it didn't before.

Reply:
I tried with that first, but it didn't work. I'm really not sure why. PKG_AUTORECONF doesn't care about target/host, right?

Reply:
Correct, autoreconf will only run once, when the package is first built for either host or target or whatever. My question is why this autoreconf change is needed now when it hasn't been needed before? Have you tested your changes with _clean_ build, are you sure you need to run autoreconf for this package?

Reply:
Yes, the issue came up when I first tried to build allwinner branch from clean git clone, i.e. the very first time I wanted to build it. I must say that before that I didn't try to build LE for a long time, so I'm not sure what was the behaviour in the past. I think that if automake 1.15 is used (without .1), no reconfiguration is needed.

If it is any help to you, I'm running latest Arch.

Reply:
@MilhouseVH 
Without reconfiguration I get this:
```
Executing (target): make 
 cd /home/jernej/repositories/LibreELEC.tv/build.LibreELEC-Allwinner.arm-9.0-devel/libtool-2.4.6 && /bin/sh /home/jernej/repositories/LibreELEC.tv/build.LibreELEC-Allwinner.arm-9.0-devel/libtool-2.4.6/build-aux/missing automake-1.15 --gnu Makefile
configure.ac:98: error: version mismatch.  This is Automake 1.15.1,
configure.ac:98: but the definition used by this AM_INIT_AUTOMAKE
configure.ac:98: comes from Automake 1.15.  You should recreate
configure.ac:98: aclocal.m4 with aclocal and run automake again.
WARNING: 'automake-1.15' is probably too old.
         You should only need it if you modified 'Makefile.am' or
         'configure.ac' or m4 files included by 'configure.ac'.
         The 'automake' program is part of the GNU Automake package:
         <http://www.gnu.org/software/automake>
         It also requires GNU Autoconf, GNU m4 and Perl in order to run:
         <http://www.gnu.org/software/autoconf>
         <http://www.gnu.org/software/m4/>
         <http://www.perl.org/>
make: *** [Makefile:1002: /home/jernej/repositories/LibreELEC.tv/build.LibreELEC-Allwinner.arm-9.0-devel/libtool-2.4.6/Makefile.in] Error 63
```
and with PKG_AUTORECONF="yes":
```
autoreconf: running: /home/jernej/repositories/LibreELEC.tv/build.LibreELEC-Allwinner.arm-9.0-devel/toolchain/bin/automake --add-missing --copy --force-missing
automake: error: cannot open < libltdl/ltdl.mk: No such file or directory
autoreconf: /home/jernej/repositories/LibreELEC.tv/build.LibreELEC-Allwinner.arm-9.0-devel/toolchain/bin/automake failed with exit status: 1
```

although it works, if in config/functions I change this line
```
export AUTORECONF="$TOOLCHAIN/bin/autoreconf --verbose --force --install -I $ACLOCAL_DIR"
```
to
```
export AUTORECONF="$TOOLCHAIN/bin/autoreconf --verbose --force -I $ACLOCAL_DIR"
```

Any ideas?

Reply:
> Yes, the issue came up when I first tried to build allwinner branch from clean git clone, i.e. the very first time I wanted to build it. 

I'll try a clean build of RPi2 overnight using this PR (without autoreconf) and see what happens on Ubuntu.

Reply:
OK I'm able to reproduce when building RPi2 once automake is bumped to 1.15.1 - after that, the `aclocal.m4` file in `libtool` (which is version 1.15) is no longer considered compatible with `automake` 1.15.1.

So yes, until `libtool` is bumped (if that ever happens) then running `autoreconf` on libtool would seem to be necessary.

While setting `PKG_AUTORECONF="yes"` should work, the reason it does not is because `intltool` isn't built until after `libtool`, and `intltool` is required to run `autoreconf` using [`config/functions`](https://github.com/LibreELEC/LibreELEC.tv/blob/master/config/functions#L373).

This can be fixed with http://sprunge.us/AILY, which ensures `intltool` is built before `libtool`.

However, since `libtool` installs `libtoolize`, and `libtoolize` is needed as part of the `autoreconf` --install process, this is a chicken/egg problem.

As you discovered, once `autoreconf` is run on `libtool` - now using the system `libtoolize` because the correct toolchain version hasn't been installed - then the file `ltdl.mk` is missing, and once again the build fails.

There are two options:

1) Patch `aclocal.m4` in `libtool` to 1.15.1, making it unecessary to run `autoreconf`:
http://sprunge.us/fRJB

2) Fix the build system so that `intltool` is built before `libtool`, and `--install` is not attempted by `autoreconf` when `libtoolize` isn't yet installed:
http://sprunge.us/gVhG

#2 is probably the correct fix, which should replace https://github.com/LibreELEC/LibreELEC.tv/pull/2088/commits/8b39c33dfc14acb2ef3334dfd17f327a265a6208 - technically they're equivalent, but #2 is using the built-in `autoreconf` procedure.

Reply:
Thanks for detailed analysis, but patch #2 doesn't work if full toolchain is already built and you just want to rebuild libtool:host. Did you try that case?

Arch, which uses same versions of autotools, does full bootstrap for libtool which is overkill IMO. Option #1 doesn't seems so bad now.

Reply:
Clean builds are a necessary evil. Please go with option #2 and clean build.